extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -621,36 +634,52 @@ public function formValidation() {
 				break;
 		}
 
-		$this->result = ( $signupForm->msgType == 'error' ) ? $signupForm->msgType : 'ok' ;
+		$this->result = ( $signupForm->msgType == 'error' ) ? $signupForm->msgType : 'ok';
 		$this->msg = $signupForm->msg;
 		$this->errParam = $signupForm->errParam;
 	}
 
 	/**
 	 * Disables User Signup Captcha for automated tests, mobile skin, and sites such as internal
-	 *
-	 * @throws MWException
 	 */
 	private function disableCaptcha() {
+		// Disable captcha for automated tests and wikia mobile and sites disabling it, e.g. Internal
+		if ( $this->shouldDisplayCaptcha() ) {
```

Comment:
I think it would be more clear if this check happened before calling `disableCaptcha`, inside the constructor. 

```php
if ( !$this->shouldDisplayCaptcha() ) {
    disableCaptcha();
}
```